File: src/main/java/com/project_task/taskid/service/TaskService.java
package com.project_task.taskid.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.project_task.taskid.entity.Task;
import com.project_task.taskid.repository.TaskRepository;

@Service
public class TaskService {

    private final TaskRepository taskRepository;

    public TaskService(TaskRepository taskRepository) {this.taskRepository = taskRepository; }

    public Task createTask(Task task) {
        return taskRepository.save(task);
    }

    public Task getTaskById(Long id) {
        return taskRepository.findById(id).orElse(null);
    }

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Task updateTask(Long id, Task taskDetails) {
        Task task = taskRepository.findById(id).orElse(null);
        if (task != null) {
            task.setTitle(taskDetails.getTitle());
            task.setDescription(taskDetails.getDescription());
            task.setStatus(taskDetails.getStatus());
            task.setDueDate(taskDetails.getDueDate());
            task.setPriority(taskDetails.getPriority());
            return taskRepository.save(task);
        }
        return null;
    }

    public void deleteTask(Long id) {
        taskRepository.deleteById(id);
    }
}

File: frontend/taskfrontend-app/src/services/Taskservice.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/tasks';

class TaskService {
    getAllTasks() {
        return axios.get(API_URL);
    }

    createTask(task) {
        return axios.post(API_URL, task);
    }

    getTaskById(taskId) {
        return axios.get(`${API_URL}/${taskId}`);
    }

    updateTask(taskId, task) {
        return axios.put(`${API_URL}/${taskId}`, task);
    }

    deleteTask(taskId) {
        return axios.delete(`${API_URL}/${taskId}`);
    }
}

export default new TaskService();
